tests/test_show_git_status_edge_cases.py

```
@@ -0,0 +1,29 @@
+from io import StringIO
+from unittest.mock import patch
+import git_commitai
+
+class TestShowGitStatusEdgeCases:
+    """Test edge cases in show_git_status."""
+
+    def test_show_git_status_exception_handling(self):
+        """Test show_git_status with exceptions."""
+        with patch("git_commitai.run_git", side_effect=Exception("Git error")):
+            with patch("sys.stdout", new=StringIO()) as fake_out:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)